training/model_management/src/azureml/model/mgmt/processors/pyfunc/convert.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""HFTransformers convert model."""
+
+import json
+import os
+import sys
+from pathlib import Path
+from typing import Dict
+
+import mlflow
+from azureml.model.mgmt.utils.common_utils import log_execution_time
+from mlflow.models.signature import ModelSignature
+from mlflow.types.schema import ColSpec, Schema
+
+from .vision.config import MLFlowSchemaLiterals, MMDetLiterals, Tasks
+
+
+def _prepare_artifacts_dict(input_dir: Path) -> Dict:
+    """Prepare artifacts dict for mlflow model.
+
+    :param input_dir: input directory
+    :type input_dir: Path
+    :return: artifacts dict
+    :rtype: Dict
+    """
+    metadata_path = os.path.join(input_dir, "model_selector_args.json")
+    with open(metadata_path) as f:
+        metadata = json.load(f)
+
+    artifacts_dict = {
+        MMDetLiterals.CONFIG_PATH: os.path.join(input_dir, metadata.get("pytorch_model_path")),
+        MMDetLiterals.WEIGHTS_PATH: os.path.join(input_dir, metadata.get("model_weights_path_or_url")),
+    }
+    return artifacts_dict
+
+
+def _get_mlflow_signature(task_type: str) -> ModelSignature:
+    """Return mlflow model signature with input and output schema given the input task type.
+
+    :param task_type: Task type used in training
+    :type task_type: str
+    :return: mlflow model signature.
+    :rtype: mlflow.models.signature.ModelSignature
+    """
+    input_schema = Schema(
+        [ColSpec(MLFlowSchemaLiterals.INPUT_COLUMN_IMAGE_DATA_TYPE, MLFlowSchemaLiterals.INPUT_COLUMN_IMAGE)]
+    )
+
+    if task_type == Tasks.MM_OBJECT_DETECTION.value:
+        output_schema = Schema(
+            [
+                ColSpec(MLFlowSchemaLiterals.OUTPUT_COLUMN_DATA_TYPE, MLFlowSchemaLiterals.OUTPUT_COLUMN_BOXES),
+            ]
+        )
+    return ModelSignature(inputs=input_schema, outputs=output_schema)
```

Comment:
Won't it fail when task type is not MM OD?

Reply:
Ideally, it should not come here if the task name is different as we have validation here - https://github.com/Azure/azureml-assets/pull/576/files/5bd6184415a0a47d4a55aac16ff585641b208a87#diff-188c7580a07a884075d0e2748ac45cae2a10ff22e6fe407e4f83d723dfb24bd9R64

I'll add the not implementation exception.